lib/fluent/plugin/filter_process_ucs_syslog.rb

```diff
@@ -240,6 +242,31 @@ def getToken(host)
       return token
     end
 
+    def updateEtcd(record)
+      sourceIp = record[ucsHostNameKey]
+
+      uri = URI.parse("http://etcd:2379/v2/keys/#{sourceIp}")
+      request = Net::HTTP::Put.new(uri)
+
+      req_options = {
+        use_ssl: uri.scheme == "https",
+      }
+
+      begin
+        response = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
+          http.request(request)
+        end
+
+        if !response.kind_of? Net::HTTPSuccess
+          log.error "Error updating etcd: Error code: #{response.code} Response: #{response.value}"
+          record["error"] += "Error updating etcd: Error code: #{response.code} Response: #{response.value}"
+        end
+      rescue SocketError => se
+        log.error "Error updating etcd: #{se.message}"
```

Comment:
Add somewhere here that the error is caused by SocketError.

Reply:
done